bundles/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/WebSocketConnection.java

```diff
@@ -349,19 +347,17 @@ Message parseIncomingMessage(byte[] data) {
                             if (idDataElements.length == 2) {
                                 payload = idDataElements[1];
                             }
-                            if (message.content.payload == null) {
+                            if (payload == null) {
                                 payload = readString(data, idx, data.length - 4 - idx);
                             }
```

Comment:
@mgeramb @Trinitus01 

Do you think this is correct? I assume that was the intended logic, because `message.content.payload` is always `null` at this place.

l. 280/282 creates the `message` object with `Message message = new Message();` which initializes `message.content` with a `content = new Content();`. In a newl created `Content` object the `payload` is `null`. `message.content.payload` is never assigned in the code between the object creation and here.